src/components/withCurrentUserPersonalDetails.tsx

```diff
@@ -34,8 +33,9 @@ export default function <TProps extends ComponentProps, TRef>(
     WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>,
 ): ComponentType<Omit<Omit<TProps, keyof HOCProps> & RefAttributes<TRef>, keyof OnyxProps>> {
     function WithCurrentUserPersonalDetails(props: Omit<TProps, keyof HOCProps>, ref: ForwardedRef<TRef>) {
+        const personalDetails = usePersonalDetails() ?? CONST.EMPTY_OBJECT;
```

Comment:
This is against https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#accessing-object-properties-and-default-values

<img width="472" alt="image" src="https://github.com/Expensify/App/assets/9059945/7c9c5e66-4ae9-4ba4-8735-0834527b9f2e">

Reply:
@eVoloshchak For TS files I think we should adhere to [these nullish-coalescing rules](https://github.com/Expensify/App/blob/main/contributingGuides/TS_STYLE.md#ts-nullish-coalescing) 👍

Here is an example that is directly line below code you've mentioned: 

```tsx
const personalDetails = usePersonalDetails() ?? CONST.EMPTY_OBJECT;
const accountID = props.session?.accountID ?? 0;
```

Reply:
> we shouldn't use async/await, [use the native Promise instead](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#esnext-are-we-allowed-to-use-insert-new-language-feature-why-or-why-not)

Answered by Jack already.

> we're calling waitForBatchedUpdatesWithAct twice, could you explain why is that needed and how does this resolve the issue? Doesn't seem right.

It was mostly trial and error process, but what I found out is that there is a race condition in the test that did not make the tests in any case deterministic (sometimes on test failed, another time two of them failed). In this case `waitForBatchedUpdatesWithAct` helps by waiting for next "tick". Jokingly, we could say that the code now is too fast :D

> was this test broken by this particular PR? If it's unrelated to this PR, we should fix it in a separate PR to avoid confusion in the future

It was caused by this PR

Hope that answers your concerns 👍